### PR TITLE
ENH: Add precomputed_slices.

### PR DESCRIPTION
**Pull request by ssanderson**

### CODE REVIEW

File: codetransformer/transformers/precomputed_slices.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""
```
Comment (llllllllll) [inline codetransformer/transformers/precomputed_slices.py:1]: maybe put this in constants?

Comment (ssanderson) [inline codetransformer/transformers/precomputed_slices.py:1]: I think this is different enough from constants to have its own file.

Comment (llllllllll) [inline codetransformer/transformers/precomputed_slices.py:1]: idk, it is about changing the constants of a function. I don't have a strong preference but I think they are related.

Comment (ssanderson) [inline codetransformer/transformers/precomputed_slices.py:1]: I think of `asconstants` as being about injecting your own custom constants.

I'd be more inclined to put this in something like a top-level `optimizers` package that has transformers that do semantics-preserving optimizaitons.

Comment (llllllllll) [inline codetransformer/transformers/precomputed_slices.py:1]: Okay, I think that we should do something like `optimizations` when we add more.

**Pull request merged**